sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_session_async.py

```diff
@@ -184,7 +185,8 @@ async def _incoming_attach(self, frame):
             self._output_handles[outgoing_handle] = new_link
             self._input_handles[frame[1]] = new_link
         except ValueError:
-            pass  # TODO: Reject link
+            # Reject Link
+            await new_link.detach()
```

Comment:
it looks like this new link is being defined in the except KeyError block, but being detached here where it wouldn't exist unless I'm missing something.

Reply:
good catch !! now its rejecting the right link :)